rust/dyn-ddlog/src/ast/debug_printing.rs

```diff
@@ -0,0 +1,290 @@
+#![cfg(feature = "debug-printing")]
```

Comment:
Meta:
- let's start adding SPDX header (possibly in an automated way) to all source files
- I realize that a lot of the code written so far is self-explanatory, still at least module-level comments would help. Another way to think of this is that we certainly want to use rustdoc to document compiler internals, so we want to document at least stuff exported from each module for `rustdoc` to generate meaningful docs.